src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityOperator.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// The record type includes synthesized '==' and '!=' operators equivalent to operators declared as follows:
+    /// 
+    /// public static bool operator==(R? r1, R? r2)
+    ///      => (object) r1 == r2 || (r1?.Equals(r2) ?? false);
```

Comment:
>(r1?.Equals(r2) ?? false) [](start = 37, length = 25)

Please consider changing this to match the implementation: `(object)r1 != null && r1.Equals(r2)`